CONTRIBUTING.md

```diff
@@ -13,3 +15,79 @@ Please read the agreement and acknowledge it by ticking the appropriate box in t
  text, for example:
 
 - [x] Tick to sign-off your agreement to the Developer Certificate of Origin (DCO) 1.1
+
+## Requirements
+
+- gradle
+- Java 1.8
+
+## Installing requirements
+
+### Java
+
+Follow the instructions for your platform.
+
+### Gradle
+
+The project uses the gradle wrapper to download the specified version of gradle.
+The gradle wrapper is run by using the following command:
+
+```bash
+$ ./gradlew
+```
+Note: on windows the command to run is gradlew.bat rather than gradlew
+
+## Coding guidelines
```

Comment:
Did you apply these guidelines to the Java ? If not suggest we drop this section for now, there's not a lot of code in this lib anyway and we can always add one and re-format later.

Reply:
I think my code does follow those guidelines, but happy to drop for now